src/track/beatmap.cpp

```diff
@@ -32,6 +34,24 @@ bool BeatLessThan(const Beat& beat1, const Beat& beat2) {
     return beat1.frame_position() < beat2.frame_position();
 }
 
+double calculateNominalBpm(const BeatList& beats, SINT sampleRate) {
+    QVector<double> beatvect;
```

Comment:
Why do we allocate a new vector and copy a subset of the elements instead of only filtering the existing BeatList?

I noticed that BeatUtils::calculateBpm() does a lot more allocations internally.

Reply:
That is done, because BeatUtils::calculateBpm was original written for the analyzer which returns a QVector().
Now this function is only used in the constructor, so the performance overheads is OK. 

Do you suggest a change here?

Reply:
No. I don't see a quick solution besides a rewrite (which is required anyway considering that all this code is far from real-time safe).